.github/workflows/windows.yml

```
@@ -413,52 +413,129 @@ jobs:
+        timeout-minutes: 5
+        if: success()
+        shell: bash
+        run: |
+          git clone --quiet --depth 1 https://github.com/microsoft/vcpkg/
+          vcpkg/bootstrap-vcpkg.sh
+
+      - name: 'vcpkg build'
+        timeout-minutes: 10
+        shell: bash
+        run: |
+          export VCPKG_ROOT="$PWD/vcpkg"
+          export CURRENT_BINARY_CACHE='cache/${{ matrix.name }}'
+          mkdir -p "$CURRENT_BINARY_CACHE"
+          export VCPKG_BINARY_SOURCES="clear;files,$GITHUB_WORKSPACE/$CURRENT_BINARY_CACHE,readwrite;"
+          vcpkg/vcpkg x-set-installed ${{ matrix.install }} --triplet=${{ matrix.arch }}-${{ matrix.plat }}
+
+      - name: 'save cache dependencies'
+        id: cache-save
+        uses: actions/cache/save@0c45773b623bea8c8e75f6c82b208c3cf94ea4f9 # v4
+        with:
+          path: 'cache/${{ matrix.name }}'
+          key: ${{ matrix.plat }}-${{ matrix.name }}-${{ hashFiles('vcpkg/installed/vcpkg/info/*', 'vcpkg/installed/vcpkg/updates/*') }}
+
       - name: 'cmake configure'
         timeout-minutes: 5
         shell: bash
         run: |
-          archgen=${{ matrix.arch }}; [ "${archgen}" = 'x86' ] && archgen='Win32'
           if [ '${{ matrix.plat }}' = 'uwp' ]; then
             system='WindowsStore'
             options='-DCMAKE_SYSTEM_VERSION=10.0'
           else
             system='Windows'
           fi
-          [ '${{ matrix.type }}' = 'Debug' ] && options+=' -DCMAKE_RUNTIME_OUTPUT_DIRECTORY_DEBUG='
-          [ '${{ matrix.type }}' = 'Release' ] && options+=' -DCMAKE_RUNTIME_OUTPUT_DIRECTORY_RELEASE='
+          export VCPKG_ROOT="$PWD/vcpkg"
           cmake -B bld ${options} \
             "-DCMAKE_SYSTEM_NAME=${system}" \
-            -DCMAKE_TOOLCHAIN_FILE=C:/vcpkg/scripts/buildsystems/vcpkg.cmake \
-            "-DCMAKE_GENERATOR_PLATFORM=${archgen}" \
+            "-DCMAKE_TOOLCHAIN_FILE=$VCPKG_ROOT/scripts/buildsystems/vcpkg.cmake" \
+            "-DVCPKG_INSTALLED_DIR=$VCPKG_ROOT/installed" \
             '-DVCPKG_TARGET_TRIPLET=${{ matrix.arch }}-${{ matrix.plat }}' \
-            -DCMAKE_VS_GLOBALS=TrackFileAccess=false \
-            '-DCMAKE_UNITY_BUILD=${{ matrix.unity }}' \
-            "-DCMAKE_C_FLAGS=${cflags}" \
             '-DCMAKE_BUILD_TYPE=${{ matrix.type }}' \
+            -DCMAKE_VS_GLOBALS=TrackFileAccess=false \
             -DCMAKE_UNITY_BUILD=ON \
             -DCURL_WERROR=ON \
             -DBUILD_EXAMPLES=ON \
             -DENABLE_WEBSOCKETS=ON \
+            -DCMAKE_RUNTIME_OUTPUT_DIRECTORY_DEBUG= \
+            -DCMAKE_RUNTIME_OUTPUT_DIRECTORY_RELEASE= \
             ${{ matrix.config }} || { cat bld/CMakeFiles/CMake*.yaml; false; }
 
+      - name: 'prepare logs on failure'
+        timeout-minutes: 5
+        if: failure()
+        shell: bash
+        run: |
+          export VCPKG_ROOT="$GITHUB_WORKSPACE/vcpkg"
+          mkdir -p bld
+          7z a -t7z -r -mx=9 logs.7z \
+            "$VCPKG_ROOT/buildtrees/*.log" \
```

```suggestion
            "$VCPKG_ROOT"/buildtrees/*.log \
```
`*` needs to stay outside of double-quotes to expand

---

It windows style, you should give all the path protected with double quotes.

---

This is bash code. Let's stick to bash syntax!

---

I think the 7z do the expansion as well.

---

I can test it on my own ci.

---

It may be able to do the expansion, but when using bash, let's do it the bash way.

---

It doing the expansion, but doing it flat. Better as you suggest.

---

It doing the same, I will do as you suggested that it will not confuse.

---

done.